tests/components/asuswrt/test_config_flow.py

```diff
@@ -235,88 +217,84 @@ async def test_on_connect_failed(hass: HomeAssistant, side_effect, error) -> Non
 
         result = await hass.config_entries.flow.async_configure(
-            flow_result["flow_id"], user_input=CONFIG_DATA
+            flow_result["flow_id"], user_input=CONFIG_DATA_TELNET
         )
         assert result["type"] == data_entry_flow.FlowResultType.FORM
         assert result["errors"] == {"base": error}
 
 
-async def test_options_flow_ap(hass: HomeAssistant) -> None:
+async def test_options_flow_ap(hass: HomeAssistant, patch_setup_entry) -> None:
     """Test config flow options for ap mode."""
     config_entry = MockConfigEntry(
         domain=DOMAIN,
-        data=CONFIG_DATA,
+        data={**CONFIG_DATA_TELNET, CONF_MODE: MODE_AP},
         options={CONF_REQUIRE_IP: True},
     )
     config_entry.add_to_hass(hass)
 
-    with PATCH_SETUP_ENTRY:
-        await hass.config_entries.async_setup(config_entry.entry_id)
-        await hass.async_block_till_done()
-        result = await hass.config_entries.options.async_init(config_entry.entry_id)
+    await hass.config_entries.async_setup(config_entry.entry_id)
+    await hass.async_block_till_done()
+    result = await hass.config_entries.options.async_init(config_entry.entry_id)
 
-        assert result["type"] == data_entry_flow.FlowResultType.FORM
-        assert result["step_id"] == "init"
-        assert CONF_REQUIRE_IP in result["data_schema"].schema
-
-        result = await hass.config_entries.options.async_configure(
-            result["flow_id"],
-            user_input={
-                CONF_CONSIDER_HOME: 20,
-                CONF_TRACK_UNKNOWN: True,
-                CONF_INTERFACE: "aaa",
-                CONF_DNSMASQ: "bbb",
-                CONF_REQUIRE_IP: False,
-            },
-        )
+    assert result["type"] == data_entry_flow.FlowResultType.FORM
+    assert result["step_id"] == "init"
+    assert CONF_REQUIRE_IP in result["data_schema"].schema
+
+    result = await hass.config_entries.options.async_configure(
+        result["flow_id"],
+        user_input={
+            CONF_CONSIDER_HOME: 20,
+            CONF_TRACK_UNKNOWN: True,
+            CONF_INTERFACE: "aaa",
+            CONF_DNSMASQ: "bbb",
+            CONF_REQUIRE_IP: False,
+        },
+    )
 
-        assert result["type"] == data_entry_flow.FlowResultType.CREATE_ENTRY
-        assert config_entry.options[CONF_CONSIDER_HOME] == 20
-        assert config_entry.options[CONF_TRACK_UNKNOWN] is True
-        assert config_entry.options[CONF_INTERFACE] == "aaa"
-        assert config_entry.options[CONF_DNSMASQ] == "bbb"
-        assert config_entry.options[CONF_REQUIRE_IP] is False
+    assert result["type"] == data_entry_flow.FlowResultType.CREATE_ENTRY
+    assert config_entry.options[CONF_CONSIDER_HOME] == 20
+    assert config_entry.options[CONF_TRACK_UNKNOWN] is True
+    assert config_entry.options[CONF_INTERFACE] == "aaa"
+    assert config_entry.options[CONF_DNSMASQ] == "bbb"
+    assert config_entry.options[CONF_REQUIRE_IP] is False
```

Comment:
Are you asserting here all options values?

If yes, please change the code to:

```python
assert config_entry.options == { CONF_CONSIDER_HOME: 20, ....(the other values) }
```

Reply:
Done